workspaces/cli-config/src/index.ts

```diff
@@ -29,8 +29,9 @@ export interface ITestingConfig {
 
 export interface IOpticTask {
```

Comment:
probably better to make this a type composed of 2 interfaces (something like type IOpticTask = IOpticStandardTaskDefinition | IOpticTestTaskDefinition). That way there will be fewer states

Reply:
Yeah I can do that — it makes more sense now that we have all the uses cases in mind. 

Overall how’s the implementation look?

Reply:
Ah I missed these types changing around IOpticTaskAliased, so I might be misreading - but to clarify: the goal for me is to not have types with states that are representable but do not or should not occur. If that can still here, then I'd consider changing the types. Your functions below should probably be https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types type guards so typescript can narrow the types for you.

Reply:
Once you do that, you should be able to remove a lot of the !s that seem to be unnecessary